### DISCUSSION TITLE
Discrete collision errors but planner is still successful

### DISCUSSION THREAD

**Discussion by dpiet** [tesseract-robotics/tesseract_planning#230] [Q&A] [answered]
What would cause a series of error messages similar to:
```
Error:   Discrete collision detected between 'eoat_link' and 'link_4' with distance -0.0379571
```
But the planner is still successful most of the time. Is there a debug I can enable somewhere or could it be a configuration problem?

**Comment by Levi-Armstrong** [comment, marked as answer]
What process taskflow are you using?

**Reply by dpiet** [reply]
We were trying out the [SnP repository](https://github.com/ros-industrial-consortium/scan_n_plan_workshop) with a different URDF and updated workcell SRDF config.

**Reply by marip8** [reply]
I think this is an artifact of the Descartes planner using OPW kinematics. The OPW solver will return solutions for all possible joint configurations for a given Cartesian waypoint (8 in total I believe); these correspond to elbow-up, elbow-down, etc. In this case, usually the elbow down configuration puts the robot in collision with the table and this error message gets generated.

For the SNP repository, you can launch the application with the argument `verbose:=true` which ironically hides these messages but should give you more information about planner failures when they occur

**Reply by dpiet** [reply]
Seems to do the trick. That explanation makes sense!
